src/cascadia/WindowsTerminal/BaseWindow.h

```diff
@@ -115,30 +114,59 @@ class BaseWindow
                 SWP_NOZORDER | SWP_NOACTIVATE);
 
             _currentDpi = uDpi;
-            NewScale(uDpi);
         }
         _inDpiChange = false;
         return 0;
     }
 
-    virtual void NewScale(UINT dpi) = 0;
-
     virtual void OnResize(const UINT width, const UINT height) = 0;
     virtual void OnMinimize() = 0;
     virtual void OnRestore() = 0;
 
-    RECT GetWindowRect() const
+    RECT GetWindowRect() const noexcept
     {
         RECT rc = { 0 };
         ::GetWindowRect(_window, &rc);
         return rc;
     }
 
-    HWND GetHandle() noexcept
+    HWND GetHandle() const noexcept
     {
         return _window;
     };
 
+    float GetCurrentDpiScale() const noexcept
+    {
+        const auto dpi = ::GetDpiForWindow(_window);
+        const auto scale = float(dpi) / float(USER_DEFAULT_SCREEN_DPI);

```

Comment:
```suggestion
        const auto scale = static_cast<float>(dpi) / static_cast<float>(USER_DEFAULT_SCREEN_DPI);
```

Using `type(x)` is a strange way to construct POD values, but it does work. It also does look strange to folks.

Reply:
@DHowett-MSFT It would be nice to have a narrow_cast so this type of cast is searchable across the codebase.  I don't think the code base uses gsl, correct?  Even if you don't want to go down that road, it would be trivial to have a user-defined one.

Reply:
@dlong11 we are a gsl codebase. Have a search for `gsl::narrow`!

Reply:
@DHowett-MSFT Thanks! I searched for gsl::narrow_cast and didn't find it.  I found gsl::narrow though!  I should have put a little more leg work into my searching.  In the example above should the recommendation be gsl::narrow_cast then?  I find these named casts to be super useful.